### ISSUE TITLE
Condition always false

### ISSUE THREAD

**Issue by maddin200** [Cisco-Talos/clamav#789]
/libclamav/regex_suffix.c line 227
Code:
  } else if (pat[*pos] == '[' && pat[*pos] == ':') {  // <- always false; maybe || required
            /* char class */
            FREE(bitmap);
            while (pat[*pos] != ']') ++*pos;
            ++*pos;

**Comment by micahsnyder** [comment]
Good catch. I wonder if that `&&` is supposed to be an `||`.  It has been this way for 14 years.  It will take some investigation to understand intent and to make the right fix.
